server/src/routers/MenuItemsRouter/parseMenuItemData.js

```diff
@@ -1,100 +1,83 @@
 import parseNutritionInfo from "./parseNutritionInfo";
+import parseIngredients from "./parseIngredients";
+
 import filters from "../../config/filters.js";
 import { E_NO_API_RES } from "../../config/constants";
 
-/**
- * Parses and combines the nutrition (by call to parseNutritionInfo),
- * filter, and ingredient data into a single json menu
- *
- * throws an error if any of the inputs are null or empty
- * or missing DATA field or empty DATA field
- *
- * @param {dict} nutritionData
- * @param {dict} filterData
- * @param {dict} ingredientData
- *
- * return {dict} with menu as a single json object
- *
- */
-
 export default function parseMenuItemData(
-    nutritionData,
-    filterData,
-    ingredientData
+    nutritionResponse,
+    filterResponse,
+    ingredientResponse
 ) {
-    // Error checking for invalid/empty data object
-    if (
-        !nutritionData ||
-        !nutritionData.DATA ||
-        !nutritionData.DATA.length ||
-        !nutritionData.DATA[0].length
-    ) {
-        throw new Error(E_NO_API_RES);
-    }
     if (
-        !filterData ||
```

Comment:
You're probably not thinking about it right now but you could refactor this using JavaScript's `Array.some()` method like so:

```
const data = [nutritionResponse, filterResponse, ingredientResponse];

const checkBadData = () => data.some(element => {
    // Checks whether the data is empty;
    return !element.data || !element.data.DATA || !element.data.DATA.length || !element.data.DATA[0].length
})
```

That's all!